hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/DFSRouterFaultInjector.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdfs.server.federation.router;
+
+import org.apache.hadoop.classification.VisibleForTesting;
+
+/**
+ * Used to inject certain faults for testing.
+ */
+public class DFSRouterFaultInjector {
+  @VisibleForTesting
+  private static DFSRouterFaultInjector instance =
+      new DFSRouterFaultInjector();
+
+  @VisibleForTesting
+  public static DFSRouterFaultInjector get() {
+    return instance;
+  }
+
+  public static void set(DFSRouterFaultInjector injector) {
+    instance = injector;
+  }
```

Comment:
Hi, @hfutatzhanghb You can set the asynchronous thread as a daemon thread, so there's no need for this class.

Reply:
Very nice suggestion, thanks. will fix it.